py/repl.c

```diff
@@ -218,7 +218,7 @@ static void print_completions(const mp_print_t *print,
     for (qstr q = q_first; q <= q_last; ++q) {
         size_t d_len;
         const char *d_str = (const char *)qstr_data(q, &d_len);
-        if (s_len <= d_len && strncmp(s_start, d_str, s_len) == 0) {
+        if (s_len <= d_len && strncmp(s_start, d_str, s_len) == 0 && d_str[0] != '_') {
```

Comment:
Please break this out into a separate PR.  It'll need changes to the tests as well.

Reply:
Split out to https://github.com/micropython/micropython/pull/17108 with a unit test (which might need some more updating to ensure the test is consistent / works across ports etc?)